nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/permissions/test/TestCommentManager.java

```diff
@@ -0,0 +1,115 @@
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentModelList;
+import org.nuxeo.ecm.core.api.security.ACE;
+import org.nuxeo.ecm.core.api.security.ACL;
+import org.nuxeo.ecm.core.api.security.ACP;
+import org.nuxeo.ecm.core.api.security.SecurityConstants;
+import org.nuxeo.ecm.core.api.security.impl.ACLImpl;
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.ecm.platform.comment.api.CommentManager;
+import org.nuxeo.ecm.platform.comment.workflow.utils.CommentsConstants;
+import org.nuxeo.ecm.platform.usermanager.UserManager;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+
+@RunWith(FeaturesRunner.class)
+@Features({ CommentFeature.class })
+public class TestCommentManager {
+
+    @Inject
+    protected CoreSession systemSession;
+
+    @Inject
+    protected CommentManager commentManager;
+
+    @Inject
+    protected CoreFeature coreFeature;
+
+    @Inject
+    protected UserManager userManager;
+
+    @Test
+    public void testCeateReadDelete() throws Exception {
+        // Create a user
+        DocumentModel userModel = userManager.getBareUserModel();
+        String schemaName = userManager.getUserSchemaName();
+        userModel.setProperty(schemaName, "username", "Foo");
+        userModel = userManager.createUser(userModel);
+
+        // Create a domain
+        DocumentModel domain = systemSession.createDocumentModel("Folder");
+        domain.setProperty("dublincore", "title", "Domain");
+        domain.setPathInfo("/", "domain");
+        domain = systemSession.createDocument(domain);
+
+        // Create a document
+        DocumentModel doc = systemSession.createDocumentModel("/domain/", "TestFile", "File");
+
+        doc = systemSession.createDocument(doc);
+
+        // Set The right ACE so Foo can read the document
+        ACP acp = doc.getACP();
+        ACE grantRead = new ACE("Foo", SecurityConstants.READ, true);
+        ACL acl = new ACLImpl();
+        acl.setACEs(new ACE[] { grantRead });
+        acp.addACL(acl);
+        doc.setACP(acp, true);
+
+        try (CoreSession userSession = coreFeature.openCoreSession("Foo")) {
+            // Get the document as the Foo user
+            DocumentModel userDoc = userSession.getDocument(doc.getRef());
+
+            // Comment the document as Foo
+            DocumentModel comment = userSession.createDocumentModel(CommentsConstants.COMMENT_DOC_TYPE);
+            comment.setProperty("comment", "text", "This is my comment");
+            commentManager.createComment(userDoc, comment);
+
+            // Check the comment document can be retrieved by Administrator via query
+            DocumentModelList dml = systemSession.query("SELECT * FROM " + CommentsConstants.COMMENT_DOC_TYPE);
```

Comment:
Just use `List<DocumentModel>` if you don't use any method from `DocumentModelList`.